geode-connectors/src/main/java/org/apache/geode/connectors/jdbc/internal/cli/CreateMappingCommand.java

```diff
@@ -300,13 +300,17 @@ public boolean updateConfigForGroup(String group, CacheConfig cacheConfig, Objec
     if (regionConfig == null) {
       return false;
     }
-
     RegionAttributesType attributes = getRegionAttribute(regionConfig);
-    addMappingToRegion(regionMapping, regionConfig);
-    if (!synchronous) {
-      createAsyncQueue(cacheConfig, attributes, queueName);
+    if (regionConfig.getType().equals("PARTITION_PROXY")
+        || regionConfig.getType().equals("REPLICATE_PROXY")) {
```

Comment:
Instead of only checking for the type on the regionConfig, I think you should also check the RegionAttributesType. On it you would check to see if the dataPolicy is empty (i.e. replicate proxy), or if the partition attributes on it has localMaxMemory of zero (i.e. partition proxy).

I was surprised to see REPLICATE_PROXY here. Did you confirm that we must set the async listener id on a replicate proxy?

Reply:
Yes. Replicated region in all members should have the same AEQ id. It's showed in code and I add a simple test case as following to verify:
@Test
  public void testAEQIdShouldBeTheSameForReplicatedRegionInAllMembers() {
    member1.invoke(() -> {
      RegionFactory<String, String> factory = cacheRule.getCache().createRegionFactory(REPLICATE);
      factory.addAsyncEventQueueId("aeq1");
      factory.create("ReplicateRegionWithAEQ");
    });

    member2.invoke(() -> {
      RegionFactory<String, String> factory = cacheRule.getCache().createRegionFactory(REPLICATE);
      factory.addAsyncEventQueueId("aeq2");
      factory.create("ReplicateRegionWithAEQ");
    });
  }
}